pd/src/state.rs

```diff
@@ -480,11 +500,27 @@ ON CONFLICT (id) DO UPDATE SET data = $1
             .map(|row| {
                 let identity_key =
                     IdentityKey::decode(row.identity_key.as_slice()).expect("db data is valid");
+                let consensus_key =
+                    tendermint::PublicKey::from_raw_ed25519(row.consensus_key.as_slice())
+                        .ok_or_else(|| anyhow::anyhow!("invalid ed25519 consensus pubkey"))?;
                 Ok(ValidatorInfo {
-                    validator: Validator::decode(row.validator_data.as_slice())?,
+                    validator: Validator {
+                        identity_key: identity_key.clone(),
+                        consensus_key: consensus_key.clone(),
+                        name: row.name.clone(),
+                        website: row.website.clone(),
+                        description: row.description.clone(),
```

Comment:
No need to clone `name`, `website`, `description`: they can be passed by reference here.